### PR TITLE
Fixed hideMessage attribute removal in server config

### LINKED ISSUE #21805: Removed hideMessage logging attribute not dynamically picked up during server configuration update
**Describe the bug**  
When a server is started with the hideMessage logging attribute set to hide messages with a specific messageID prefix, and when the server.xml is updated to remove that hideMessage logging attribute (indicating all messages should be shown), the configuration is not updated and still shows that the messageID that was set previously in the hideMessage logging attribute should be hidden.

**Steps to Reproduce**  
1. Start server with the following logging configuration in the server.xml:
`<logging maxFiles="10" hideMessage="CWWKZ0058I" />`
2. Observe in the messages.log/console.log files for the following message:
`TRAS3001I: The following messages are hidden from the console.log and messages.log files: [CWWKZ0058I]`
3. Verify that the `CWWKZ0058I` message is not shown in the messages.log/console.log files.
4. Update the server.xml and remove the hideMessage logging attribute, as follows:
`<logging maxFiles="10" />`
5. The messages.log/console.log files will still show the following message:
`TRAS3001I: The following messages are hidden from the console.log and messages.log files: [CWWKZ0058I]`
6. The `CWWKZ0058I` message is still hidden in the messages.log/console.log files

**Expected behavior**  
After step 4, the `TRAS3001I` message should NOT be shown, and the `CWWKZ0058I` message should NOT be hidden in the messages.log and console.log files.

**Diagnostic information:**  
 - OpenLiberty Version: 22.0.0.7

### PR DESCRIPTION
**Pull request by pgunapal**
fixes #21805

- Made sure to save the initValue of the hideMessages attribute (empty string), during server initialization, hence when the hideMessage attribute is removed from server.xml, the hideMessage is set to an empty string, instead of the previous values.

### CODE REVIEW

**Comment by pgunapal** [comment]
#build

**Comment by pgunapal** [comment]
#build

**Comment by pgunapal** [comment]
#build

File: dev/com.ibm.ws.logging_fat/publish/files/server-hidemsgattrbremoval.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<!--
```
Comment (Channyboy) [inline dev/com.ibm.ws.logging_fat/publish/files/server-hidemsgattrbremoval.xml]: Camel Case file name

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/TestHideMessages.java [hunk 2]
```diff
@@ -34,20 +35,29 @@ public class TestHideMessages {
 
     private static LibertyServer msgServer = LibertyServerFactory.getLibertyServer("com.ibm.ws.logging.hidemessage");
     private static final Class<?> logClass = TestHideMessages.class;
-    static String TWO_HIDDEDN_MESSAGE_SERVER = "server-twomessageids.xml";
+    static String TWO_HIDDEN_MESSAGE_SERVER = "server-twomessageids.xml";
```
Comment (Channyboy) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/TestHideMessages.java]: might as well fix the camel casing of the `server-twomessageids` aswell

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/TestHideMessages.java [hunk 3]
```diff
@@ -87,8 +98,63 @@ public void testDynamicAddMessageIds() throws Exception {
         Log.info(logClass, name.getMethodName(), "Exiting test " + name.getMethodName());
     }
 
-    @AfterClass
-    public static void completeTest() throws Exception {
+    // Tests when the hideMessage attribute is removed from the server.xml, no messages should be hidden.
```
Comment (Channyboy) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/TestHideMessages.java]: Use `/**` to describe (test) method

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/TestHideMessages.java [hunk 4]
```diff
@@ -87,8 +98,63 @@ public void testDynamicAddMessageIds() throws Exception {
...
+        assertTrue("Hidden Message CWWKF0012I should be seen in console.log",
+                   msgServer.findStringsInLogs("CWWKF0012I:", msgServer.getMatchingLogFile("console.log")).size() > 0);
+        assertTrue("Info message about hidden messageID prefixes should not be logged",
+                   msgServer.findStringsInLogsUsingMark("TRAS3001I:", msgServer.getMatchingLogFile("messages.log")).isEmpty());
+
+        Log.info(logClass, name.getMethodName(), "Exiting test " + name.getMethodName());
+
+    }
+
+    // Tests when the hideMessage attribute is set to an empty string from the server.xml, no messages should be hidden.
```
Comment (Channyboy) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/TestHideMessages.java]: Use `/**` to describe (test) method

File: dev/com.ibm.ws.logging_fat/publish/files/server-hidemsgattrbremoval.xml (hunk 1 above)
Comment (pgunapal) [reply dev/com.ibm.ws.logging_fat/publish/files/server-hidemsgattrbremoval.xml]: Fixed.

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/TestHideMessages.java (hunk 2 above)
Comment (pgunapal) [reply dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/TestHideMessages.java]: Fixed.

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/TestHideMessages.java (hunk 3 above)
Comment (pgunapal) [reply dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/TestHideMessages.java]: Fixed.

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/TestHideMessages.java (hunk 4 above)
Comment (pgunapal) [reply dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/TestHideMessages.java]: Fixed.

**Review by Channyboy — APPROVED** [review]
lgtm

**Comment by pgunapal** [comment]
#build

**Comment by pgunapal** [comment]
The failing tests are known test failures in unrelated components, that have defects already opened for.

**Comment by pgunapal** [comment]
#libby

**Pull request merged**